File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/EvidenceVariable.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class EvidenceVariable : DomainResource, IResource
{
    public string? Description { get; set; }
    public string? Date { get; set; }
    public Base.ContactDetail[]? Endorser { get; set; }
    public string? Publisher { get; set; }
    public string? ApprovalDate { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public string? Name { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public string? Copyright { get; set; }
    public string? Type { get; set; }
    public Base.CodeableConcept[]? Topic { get; set; }
    public string? Title { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public Base.ContactDetail[]? Author { get; set; }
    public required EvidenceVariable_Characteristic[] Characteristic { get; set; }
    public required string Status { get; set; }
    public string? Subtitle { get; set; }
    public string? Url { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? LastReviewDate { get; set; }
    public Base.ContactDetail[]? Editor { get; set; }
    public Base.ContactDetail[]? Reviewer { get; set; }
    public string? ShortTitle { get; set; }
    public string? Version { get; set; }
    public Base.RelatedArtifact[]? RelatedArtifact { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }
    public Base.Period? EffectivePeriod { get; set; }

    public class EvidenceVariable_Characteristic : BackboneElement
    {
        public string? Description { get; set; }
        public object? Definition    
        {
            get
            {
                if (DefinitionExpression is not null)
                {
                    return DefinitionExpression;
                }
        
                if (DefinitionDataRequirement is not null)
                {
                    return DefinitionDataRequirement;
                }
        
                if (DefinitionTriggerDefinition is not null)
                {
                    return DefinitionTriggerDefinition;
                }
        
                if (DefinitionCanonical is not null)
                {
                    return DefinitionCanonical;
                }
        
                if (DefinitionReference is not null)
                {
                    return DefinitionReference;
                }
        
                if (DefinitionCodeableConcept is not null)
                {
                    return DefinitionCodeableConcept;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.ResourceExpression))
                {
                    DefinitionExpression = (Base.ResourceExpression)value;return;
                }
        
                if (value?.GetType() == typeof(Base.DataRequirement))
                {
                    DefinitionDataRequirement = (Base.DataRequirement)value;return;
                }
        
                if (value?.GetType() == typeof(Base.TriggerDefinition))
                {
                    DefinitionTriggerDefinition = (Base.TriggerDefinition)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    DefinitionCanonical = (string)value;return;
                }
        
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    DefinitionReference = (Base.ResourceReference)value;return;
                }
        
                if (value?.GetType() == typeof(Base.CodeableConcept))
                {
                    DefinitionCodeableConcept = (Base.CodeableConcept)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public bool? Exclude { get; set; }
        public string? GroupMeasure { get; set; }
        public Base.ResourceExpression? DefinitionExpression { get; set; }
        public string? TimeFromStart { get; set; }
        public string? ParticipantEffectiveDuration { get; set; }
        public Base.DataRequirement? DefinitionDataRequirement { get; set; }
        public Base.TriggerDefinition? DefinitionTriggerDefinition { get; set; }
        public string? DefinitionCanonical { get; set; }
        public Base.ResourceReference? DefinitionReference { get; set; }
        public string? ParticipantEffectiveTiming { get; set; }
        public string? ParticipantEffectiveDateTime { get; set; }
        public Base.Period? ParticipantEffectivePeriod { get; set; }
        public Base.CodeableConcept? DefinitionCodeableConcept { get; set; }
        public object? ParticipantEffective    
        {
            get
            {
                if (ParticipantEffectiveDuration is not null)
                {
                    return ParticipantEffectiveDuration;
                }
        
                if (ParticipantEffectiveTiming is not null)
                {
                    return ParticipantEffectiveTiming;
                }
        
                if (ParticipantEffectiveDateTime is not null)
                {
                    return ParticipantEffectiveDateTime;
                }
        
                if (ParticipantEffectivePeriod is not null)
                {
                    return ParticipantEffectivePeriod;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    ParticipantEffectiveDuration = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ParticipantEffectiveTiming = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ParticipantEffectiveDateTime = (string)value;return;
                }
        
                if (value?.GetType() == typeof(Base.Period))
                {
                    ParticipantEffectivePeriod = (Base.Period)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.UsageContext[]? UsageContext { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/EvidenceVariable.py
from typing import Optional, List
from base import UsageContext
from base import Annotation
from base import Period
from base import ContactDetail
from base import DataRequirement
from base import CodeableConcept
from base import TriggerDefinition
from base import Expression
from base import RelatedArtifact
from base import Timing
from base import Duration
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class EvidenceVariable_Characteristic(BackboneElement):
    description: Optional[str] = None
    exclude: Optional[bool] = None
    group_measure: Optional[str] = None
    definition_expression: Optional[Expression] = None
    time_from_start: Optional[Duration] = None
    participant_effective_duration: Optional[Duration] = None
    definition_data_requirement: Optional[DataRequirement] = None
    definition_trigger_definition: Optional[TriggerDefinition] = None
    definition_canonical: Optional[str] = None
    definition_reference: Optional[Reference] = None
    participant_effective_timing: Optional[Timing] = None
    participant_effective_date_time: Optional[str] = None
    participant_effective_period: Optional[Period] = None
    definition_codeable_concept: Optional[CodeableConcept] = None
    usage_context: Optional[List[UsageContext]] = None

class EvidenceVariable(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    endorser: Optional[List[ContactDetail]] = None
    publisher: Optional[str] = None
    approval_date: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    name: Optional[str] = None
    use_context: Optional[List[UsageContext]] = None
    copyright: Optional[str] = None
    type: Optional[str] = None
    topic: Optional[List[CodeableConcept]] = None
    title: Optional[str] = None
    note: Optional[List[Annotation]] = None
    author: Optional[List[ContactDetail]] = None
    characteristic: list[EvidenceVariable_Characteristic] = []
    status: str
    subtitle: Optional[str] = None
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    last_review_date: Optional[str] = None
    editor: Optional[List[ContactDetail]] = None
    reviewer: Optional[List[ContactDetail]] = None
    short_title: Optional[str] = None
    version: Optional[str] = None
    related_artifact: Optional[List[RelatedArtifact]] = None
    contact: Optional[List[ContactDetail]] = None
    effective_period: Optional[Period] = None